### PR TITLE
feat(demo): Footer harmonised, use-cases, mandatory, optional knobs - FRONT-822

### PR DESCRIPTION
**Pull request by planctus**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js [hunk 1]
```diff
@@ -1,52 +1,412 @@
...
-    notes: { markdown: notes, json: dataGroup2 },
-  })
-  .add('group 3', () => footerHarmonised(dataGroup3), {
-    notes: { markdown: notes, json: dataGroup3 },
-  });
+  .addDecorator(withKnobs)
+  .add(
+    'group 1',
+    () => {
+      button('With our without class names', classBtnToggler, useCasesGroup);
```
Comment (papegaill) [inline src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js]: with **or** without?

File: src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js [hunk 2]
```diff
@@ -1,52 +1,412 @@
...
+const dataG3 = [...dataGroup3.sections];
+// Labels for the buttons.
+const requiredGroupId = 'Mandatory elements';
+const optionalGroupId = 'Optional elements';
+const useCasesGroup = 'Use cases';
+
+// Buttons callbacks for optional elements.
+// Group 3
+const partnershipLogoBtnToggler = () => {
+  dataG3[0].logos[0] = dataG3[0].logos[0] ? false : logo0;
```
Comment (papegaill) [inline src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js]: Whitout partnershipLogo we still have `<li>` with empty `` tags
```
<ul class="ecl-footer-harmonised__logo-list">
  <li class="ecl-footer-harmonised__logo-item"><img src="" alt="" title="" class="ecl-footer-harmonised__logo" /></li>
  <li class="ecl-footer-harmonised__logo-item"><img src="" alt="" title="" class="ecl-footer-harmonised__logo" /></li>
  <li class="ecl-footer-harmonised__logo-item"><img src="static/media/logo--en.30b933cc.svg" alt="European Commission logo" title="European Commission" class="ecl-footer-harmonised__logo" /></li>
</ul>
```
Isn't it better to really delete logo objects?

Comment (planctus) [reply src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js]: It is. :)

File: src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js (hunk 1 above)
Comment (planctus) [reply src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js]: sure.. ;)

**Pull request closed without merging**